Author: Ole Streicher <olebole@debian.org>
Description: Don't call home for logging
 By default, Aladin loggs back every start, which can be disabled in the
 configuration. For a better privacy, we reverse this behaviour.
 .
 Note however, that Aladin is an online tool and automatically connects to
 GLU server to retrieve the list of available data sources. This is intended
 behaviour of the program, which would otherwise not function as expected.
Index: aladin/cds/aladin/Configuration.java
===================================================================
--- aladin.orig/cds/aladin/Configuration.java	2016-12-19 11:51:39.513746219 +0100
+++ aladin/cds/aladin/Configuration.java	2016-12-19 12:01:35.301733992 +0100
@@ -845,7 +845,7 @@
    /** Retourne true si le mode log est activé */
    protected boolean isLog() {
       String s = get(LOG);
-      return s==null || s.equals(ACTIVATED);
+      return s!=null && s.equals(ACTIVATED);
    }
 
    /** Retourne true s'il faut un slider d'époque */
@@ -1736,7 +1736,7 @@
       if( ocache==null || !s1.equals(ocache) ) set(MAXCACHE,s1);
 
       String s = get(LOG);
-      if( s!=null && s.equals(ACTIVATED) ) remove(LOG);
+      if( s==null ) set(LOG, NOTACTIVATED);
 
       s = get(HELP);
       if( s!=null && s.equals(ACTIVATED) ) remove(HELP);
@@ -1988,14 +1988,12 @@
       // Pourle log
       if( logChoice!=null ) {
          if( logChoice.getSelectedIndex()==1 ) {
-            if( get(LOG)==null ) aladin.glu.log("Log", "off");
-            set(LOG,(String)logChoice.getSelectedItem());
+	     if( isLog() ) aladin.glu.log("Log", "off");
          }
          else {
-            boolean pub = get(LOG)!=null && get(LOG).equals(NOTACTIVATED);
-            remove(LOG);
-            if( pub ) aladin.glu.log("Log", "on");
+	     if( !isLog() ) aladin.glu.log("Log", "on");
          }
+	 set(LOG,(String)logChoice.getSelectedItem());
       }
 
       // Pour l'assistant débutant
Index: aladin/Aladin0.string
===================================================================
--- aladin.orig/Aladin0.string	2016-12-19 11:51:39.485746219 +0100
+++ aladin/Aladin0.string	2016-12-19 12:01:35.305733992 +0100
@@ -1665,7 +1665,7 @@
 UPCACHE	Permanent disk cache
 UPCACHEH	Aladin requires local disk space for keeping data from one session to the other. More it has, better it runs. It is localized in your $HOME/.aladin directory.
 UPLOG CDS log
-UPLOGH You will help the CDS team to improve Aladin by keeping the log mechanism activated. These "logs" are anonymous and do not transmit any local information (paths, file names,...) just some usage statistics and possibly bug reports. Thanks !
+UPLOGH You will help the CDS team to improve Aladin by activating the log mechanism. These "logs" are anonymous and do not transmit any local information (paths, file names,...) just some usage statistics and possibly bug reports. Thanks !
 UPHELP Wizard
 UPHELPH Switch on/off the beginner help. Help new Aladin users to discover progressively the Aladin graphical interface.
 #UPTAGCENTER	Centered tag
